src/dodal/plans/verify_undulator_gap.py

```diff
@@ -0,0 +1,70 @@
+from dodal.devices.DCM import DCM
+from dodal.devices.undulator import Undulator, UndulatorGapAccess
+from dodal.log import LOGGER
+
+
+def _get_energy_to_distance_table(lookup_table_path: str) -> dict[float, float]:
+    energy_to_distance_table: dict[float, float] = {}
+    with open(lookup_table_path, mode="r") as table:
+        table_start = False
+        for line in table:
+            line = line.strip()
+            if line.startswith("Units"):
+                table_start = True
+                continue
+            if table_start:
+                value = line.split()
+                energy_to_distance_table[float(value[0])] = float(value[1])
+    return energy_to_distance_table
+
+
+def _get_closest_gap_to_dcm_energy(
+    dcm_energy: float, energy_to_distance_table: dict[float, float]
+) -> float:
+    table_energies = energy_to_distance_table.keys()
+    table_energy_value_to_use: float = 0
+    closest_distance_to_energy_from_table = dcm_energy
+    for energy in table_energies:
+        distance = abs(dcm_energy - energy)
+        if distance < closest_distance_to_energy_from_table:
+            closest_distance_to_energy_from_table = distance
+            table_energy_value_to_use = energy
+    return energy_to_distance_table[table_energy_value_to_use]
+
+
+def verify_undulator_gap(undulator: Undulator, DCM: DCM) -> bool:
```

Comment:
Should: If this is a plan then we need to use `yield from bps.set` etc. throughout rather than plain old `.set()`. I'm not sure if it should be though, I wonder if it makes more sense to just have an `energy` signal on the undulator that when set does this, it will make the waiting on it to change nicer too. Alternatively we could have a `UndulatorDCM` composite device that we change energy on? @dperl-dls, @olliesilvester and @rtuck99 shall we have a chat about this?

Reply:
Should: I also don't really like the name `verify` as it implies that we're just checking it. We're not because we're also moving it. I would just call it `set_undulator_gap_for_dcm_energy`

Reply:
I like the idea of an UndulatorDCM composite device with an energy signal, that sounds clean to me, it would be great to be able to `yield from bps.abs_set(energy, 12700)`

Reply:
Yes, I think this is what we will go with